craterpy/tests/test_classes.py

```diff
@@ -97,3 +146,363 @@ def test_antimeridian_splitting(self):
+        original_cdb = CraterDatabase(df)
+        
+        # Write to GeoJSON
+        import tempfile
+        import os
+        
+        with tempfile.NamedTemporaryFile(suffix='.geojson', delete=False) as tmp:
+            tmp_path = tmp.name
+        
+        try:
+            # Save to file
+            original_cdb.to_geojson(filename=tmp_path)
+            
+            # Read back the file
+            imported_cdb = CraterDatabase.read_shapefile(tmp_path)
+            
+            # Verify data was preserved
+            self.assertEqual(len(imported_cdb.data), len(original_cdb.data))
+            self.assertEqual(imported_cdb.data.shape[0], 2)
+            # Check key attributes
+            np.testing.assert_array_almost_equal(imported_cdb.lat.values, original_cdb.lat.values)
+            np.testing.assert_array_almost_equal(imported_cdb.lon.values, original_cdb.lon.values)
+            np.testing.assert_array_almost_equal(imported_cdb.rad.values, original_cdb.rad.values)
+        finally:
+            # Clean up
+            if os.path.exists(tmp_path):
+                os.remove(tmp_path)
+
+    def test_to_geojson_with_custom_columns(self):
+        """Test reading a shapefile with non-standard column names."""
+        # Create a GeoDataFrame with non-standard column names
+        import geopandas as gpd
+        
+        gdf = gpd.GeoDataFrame({
+            "latitude_deg": [0.0, 10.0],
+            "longitude_deg": [0.0, 20.0],
+            "diameter_km": [2.0, 4.0],  # Diameters in km
+            "crater_name": ["Crater A", "Crater B"],
+            "geometry": [Point(0, 0), Point(20, 10)]
+        }, crs="EPSG:4326")
+        
+        import tempfile
+        import os
+        
+        with tempfile.NamedTemporaryFile(suffix='.geojson', delete=False) as tmp:
+            tmp_path = tmp.name
+        
+        try:
+            # Save to file with non-standard column names
+            gdf.to_file(tmp_path, driver="GeoJSON")
+            
+            # Read the file with non-standard column names
+            imported_cdb = CraterDatabase.read_shapefile(tmp_path, units="km")
+            
+            # Verify data was preserved and correctly interpreted
+            self.assertEqual(len(imported_cdb.data), 2)
+            
+            # Check that our read_shapefile method correctly identified the columns
+            np.testing.assert_array_almost_equal(imported_cdb.lat.values, [0.0, 10.0])
+            np.testing.assert_array_almost_equal(imported_cdb.lon.values, [0.0, 20.0])
+            
+            # Since diameter was converted to radius in km and then to meters,
+            # we expect: diameter_km / 2 * 1000 = [1000.0, 2000.0]
+            np.testing.assert_array_almost_equal(imported_cdb.rad.values, [1000.0, 2000.0])
+        finally:
+            # Clean up
+            if os.path.exists(tmp_path):
+                os.remove(tmp_path)
+
+    def test_read_shapefile_with_body_and_units(self):
```

Comment:
not needed after changes above. other read_geojson tests might be able to be simpler and more isolated too (most column handling should happen in the constructor)